x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java

```diff
@@ -304,10 +304,16 @@ private static String randomRoleDescriptors(boolean includeRemoteIndices) {
     }
 
     boolean nodeSupportApiKeyRemoteIndices(Map<String, Object> nodeDetails) {
-        // TODO[lor]: the method can be kept, but we need to replace version check with features checks
-        String versionString = (String) nodeDetails.get("version");
-        Version version = Version.fromString(versionString.replace("-SNAPSHOT", ""));
-        return version.onOrAfter(API_KEY_SUPPORT_REMOTE_INDICES_VERSION);
+        var nodeVersion = nodeDetails.get("version");
```

Comment:
Could we use more explicit types here, to make it clearer what types we're working with here?